src/linear-algebra.lisp

```diff
@@ -23,18 +23,28 @@
   "A representation of an operator on a quantum state. This will be a unitary square matrix where each dimension is a power-of-two."
   `(simple-array cflonum (,n ,n)))
 
+;;; This is intended to be a shorthand for testing.
+(defun wf (&rest elts)
+  "Construyct a wavefunction from the elements ELTS."
+  (declare (dynamic-extent elts))
```

Comment:
> In some containing form, F, this declaration asserts for each vari (which need not be bound by F), and for each value vij that vari takes on, and for each object xijk that is an otherwise inaccessible part of vij at any time when vij becomes the value of vari, that just after the execution of F terminates, xijk is either inaccessible (if F established a binding for vari) or still an otherwise inaccessible part of the current value of vari (if F did not establish a binding for vari). The same relation holds for each fni, except that the bindings are in the function namespace.

🤕

Reply:
haha

Reply:
I really cannot make sense of that paragraph, and the examples don't help either. Do you have a better example/description of `dynamic-extent`?

Reply:
I also cannot decipher the above hyperspec legalese, but my limited understanding is that `dynamic-extent` means that the object / variable is not ["accessible"](http://clhs.lisp.se/Body/26_glo_a.htm#accessible) (i.e. can't/won't be referenced) outside of the "dynamic extent" of the declaring form / function. I.e. it doesn't "escape". In this case, once the function exits, we declare that that nothing else can "see" it, and therefore the compiler is free to take this into account for optimization.

Usually, the intent is to tell the compiler that it's safe to stack-allocate the thing.

Reply:
>Usually, the intent is to tell the compiler that it's safe to stack-allocate the thing.
Right. (Never get the declaration wrong!)